### PR TITLE
Update `retention_in_days` config to fix #99

### LINKED ISSUE #99: Log analytics retention period always set to 30
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Versions

**terraform**: 0.15.1

**azure provider**: 2.56.0

**module**: 0.3.1

### Description

**Describe the bug**
When changing the value of input variable `configure_management_resources.log_analytics.config.retention_in_days`, the module will correctly configure the `Deploy-Log-Analytics` Policy Assignment, but the Log Analytics workspace remains at the default value of `30` days.

**Steps to Reproduce**

1. Configure a default deployment using the code in the example below, noting that the value of input variable `configure_management_resources.log_analytics.config.retention_in_days` has been set to `60`
2. Run the deployment
3. Validate the retention settings on the `Deploy-Log-Analytics` Policy Assignment
4. Validate the retention settings on the Log Analytics workspace

**Code Sample**

```hcl
data "azurerm_client_config" "current" {
}

module "enterprise_scale" {
  source  = "Azure/caf-enterprise-scale/azurerm"
  version = "0.3.1"

  root_parent_id = data.azurerm_client_config.current.tenant_id
  root_id        = "sn"
  root_name      = "Log Analytics Test"

  }

  subscription_id_overrides = {
    management = [data.azurerm_client_config.current.subscription_id]
  }

  deploy_management_resources    = true
  configure_management_resources = local.configure_management_resources
  subscription_id_management     = data.azurerm_client_config.current.subscription_id

}

locals {
  configure_management_resources = {
    settings = {
      log_analytics = {
        enabled = true
        config = {
          retention_in_days                           = 60
          enable_monitoring_for_arc                   = true
          enable_monitoring_for_vm                    = true
          enable_monitoring_for_vmss                  = true
          enable_solution_for_agent_health_assessment = true
          enable_solution_for_anti_malware            = true
          enable_solution_for_azure_activity          = true
          enable_solution_for_change_tracking         = true
          enable_solution_for_service_map             = true
          enable_solution_for_sql_assessment          = true
          enable_solution_for_updates                 = true
          enable_solution_for_vm_insights             = true
          enable_sentinel                             = true
        }
      }
      security_center = {
        enabled = true
        config = {
          email_security_contact             = "security_contact@replace_me"
          enable_defender_for_acr            = true
          enable_defender_for_app_services   = true
          enable_defender_for_arm            = true
          enable_defender_for_dns            = true
          enable_defender_for_key_vault      = true
          enable_defender_for_kubernetes     = true
          enable_defender_for_servers        = true
          enable_defender_for_sql_servers    = true
          enable_defender_for_sql_server_vms = true
          enable_defender_for_storage        = true
        }
      }
    }
    location = null
    tags     = null
    advanced = null
  }
}
```

**Screenshots**

Correctly configured `retentionInDays` parameter value on the `Deploy-Log-Analytics` Policy Assignment:

image

Incorrectly configured `Data Retention (Days)` setting on the Log Analytics workspace:

image

**Additional context**

### PR DESCRIPTION
**Pull request by krowlandson**
This PR introduces a change to the logic used to set the Log Analytics workspace setting for `retention_in_days` to fix issue #99 

> **NOTE:** This is not a breaking change, but will result in an update to customer environments where this value has been modified in code but incorrectly applied in Azure,

### CODE REVIEW

**Comment by krowlandson** [comment]
## Attaching evidence

- Terraform plan after changing the value on an existing deployment

image

- Resultant change visible in the Azure Portal

image

File: modules/management/locals.tf [hunk 1]
```diff
@@ -1,8 +1,6 @@
 # The following block of locals are used to avoid using
 # empty object types in the code.
 locals {
-  empty_list   = []
```
Comment (krowlandson) [inline modules/management/locals.tf:4]: Removed as not required

File: modules/management/locals.tf [hunk 2]
```diff
@@ -1,8 +1,6 @@
...
-  empty_map    = {}
```
Comment (krowlandson) [inline modules/management/locals.tf:5]: Removed as not required

**Review by sblair01 — APPROVED** [review]
Looks good to me

**Pull request merged**